### PR TITLE
Target user id

### PR DESCRIPTION
**Pull request by stephschuster**

### CODE REVIEW

File: packages/engine.io-client/src/stellarSocket.js [hunk 1]
```diff
@@ -114,17 +116,17 @@ function stellarSocketFactory(eio) {
         }
       });
     },
-    _doConnect(url, { userId, token, secure, tokenType, eioConfig = { upgrade: true, rememberUpgrade: true } }) {
+    _doConnect(url, { userId, token, secure, tokenType, params, eioConfig = { upgrade: true, rememberUpgrade: true } }) {
       log.info(`@StellarEngineIO._doConnect: ${userId}, ${token}`);
       return new Promise((resolve, reject) => {
         this.state = 'connecting';
-
+        const urlParams = _.assign({}, params,
```
Comment (sloops77) [inline packages/engine.io-client/src/stellarSocket.js]: personally i prefer the more concise way of:
```
const urlParams = assign({ 'x-auth-user': userId, 'x-auth-token': token, 'x-auth-token-type': tokenType },   params);

File: packages/engine.io-client/src/stellarSocket.js [hunk 2]
```diff
@@ -2,6 +2,8 @@
  * Created by arolave on 15/06/2017.
  */
 import Promise from 'bluebird';
+import qs from 'qs';
+import _ from 'lodash';
```
Comment (sloops77) [inline packages/engine.io-client/src/stellarSocket.js]: import only the assign function
`import assign from 'lodash/assign';`

File: packages/engine.io-client/package.json [hunk 3]
```diff
@@ -30,11 +30,14 @@
     "eslint-config-airbnb-base": "^11.2.0",
     "eslint-plugin-import": "^2.3.0",
     "jest": "^20.0.4",
-    "rimraf": "^2.6.1"
+    "rimraf": "^2.6.1",
+    "webpack": "2.5.1"
```
Comment (sloops77) [inline packages/engine.io-client/package.json]: you shouldnt need the webpack import

File: packages/engine.io-client/src/stellarSocket.js [hunk 4]
```diff
@@ -114,17 +116,17 @@ function stellarSocketFactory(eio) {
...
       });
     },
-    _doConnect(url, { userId, token, secure, tokenType, eioConfig = { upgrade: true, rememberUpgrade: true } }) {
+    _doConnect(url, { userId, token, secure, tokenType, params, eioConfig = { upgrade: true, rememberUpgrade: true } }) {
       log.info(`@StellarEngineIO._doConnect: ${userId}, ${token}`);
       return new Promise((resolve, reject) => {
         this.state = 'connecting';
-
+        const urlParams = _.assign({}, params,
+            { 'x-auth-user': userId, 'x-auth-token': token, 'x-auth-token-type': tokenType });
```
Comment (sloops77) [inline packages/engine.io-client/src/stellarSocket.js]: can you add a test to `bridge/tests/clientserver.test.js` for url parameter sending

File: packages/engine.io-client/package.json (hunk 3 above)
Comment (ozsay) [inline packages/engine.io-client/package.json]: we couldn't run `yarn add qs` without it

File: packages/engine.io-client/src/stellarSocket.js [hunk 5]
```diff
@@ -2,6 +2,8 @@
  * Created by arolave on 15/06/2017.
  */
 import Promise from 'bluebird';
+import qs from 'qs';
+import assign from 'lodash';
```
Comment (sloops77) [inline packages/engine.io-client/src/stellarSocket.js]: `import assign from 'lodash/assign'`

File: packages/engine.io-client/src/stellarSocket.js [hunk 6]
```diff
@@ -114,17 +116,17 @@ function stellarSocketFactory(eio) {
         }
       });
     },
-    _doConnect(url, { userId, token, secure, tokenType, eioConfig = { upgrade: true, rememberUpgrade: true } }) {
+    _doConnect(url, { userId, token, secure, tokenType, params, eioConfig = { upgrade: true, rememberUpgrade: true } }) {
       log.info(`@StellarEngineIO._doConnect: ${userId}, ${token}`);
       return new Promise((resolve, reject) => {
         this.state = 'connecting';
-
+        const urlParams =
```
Comment (sloops77) [inline packages/engine.io-client/src/stellarSocket.js]: why a new line?

File: packages/engine.io-client/src/stellarSocket.js [hunk 7]
```diff
@@ -114,17 +116,17 @@ function stellarSocketFactory(eio) {
...
+            assign({ 'x-auth-user': userId, 'x-auth-token': token, 'x-auth-token-type': tokenType }, params);
         let socketAttempt = null;
         try {
           socketAttempt = new eio.Socket(
```
Comment (sloops77) [inline packages/engine.io-client/src/stellarSocket.js]: can you simplify to one line?

**Pull request merged**